CMakeLists.txt

```diff
@@ -1,17 +1,10 @@
-cmake_minimum_required(VERSION 3.10.0)
+cmake_minimum_required(VERSION 3.24.0)
```

Comment:
Needed to bump min cmake version as the current was already older than some of the features we were using